### ISSUE TITLE
Accessibility enhancements for p-autocomplete

### ISSUE THREAD

**Issue by sandipchitale** [primefaces/primeng#8743]
**I'm submitting a ...**  (check one with "x")
```
[X] bug report => Search github for a similar issue or PR before submitting
[ ] feature request => Please check if request is not on the roadmap already https://github.com/primefaces/primeng/wiki/Roadmap
[ ] support request => Please do not submit support request here, instead see http://forum.primefaces.org/viewforum.php?f=35
```

**Plunkr Case (Bug Reports)**
Please demonstrate your case at stackblitz by using the issue template below. Issues without a test case have much less possibility to be reviewd in detail and assisted.

https://stackblitz.com/github/primefaces/primeng-issue-template

**Current behavior**
Use p-autocomplete. Run Chrome Devtools Accessibility Audit.

**Expected behavior**
Accessibility report from Accessibility Audit in Chrome Devtools console should be clean.

**Minimal reproduction of the problem with instructions**

- Create an app with a p-autocomplete component or even use the PrimeNG component sample page 

- Open Chrome Devtools and run accessibility audits

The following two isseus happen.

1. role=searchbox and aria-expanded are not compatible. See: https://www.w3.org/TR/wai-aria-1.1/#searchbox . role=list may work with aria-expanded.

2. The aria-controls should be added to the input only when the drop down div is shown.
3. The aria-activedescendant should be added only when the when the drop down div is shown and one of the items is highlighted.

You can use the Axe extension to see the dynamic behavior when the dropdown is shown.

I may file a pull request.

**What is the motivation / use case for changing the behavior?**

To get clean accessibility report.

**Please tell us about your environment:**
yarn, angular 9

* **Angular version:** 5.X

9.x
* **PrimeNG version:** 5.X

9.0.3

* **Browser:** [all | Chrome XX | Firefox XX | IE XX | Safari XX | Mobile Chrome XX | Android X.X Web Browser | iOS XX Safari | iOS XX UIWebView | iOS XX WKWebView ]

all

* **Language:** [all | TypeScript X.X | ES6/7 | ES5]
all

* **Node (for AoT issues):** `node --version` =   
12.x

**Comment by sandipchitale** [comment]
I could not build so did not submit a PR.
But here are the diffs that seem to fix the issue:

```
-            [ngClass]="'ui-inputtext ui-widget ui-state-default ui-corner-all ui-autocomplete-input'" [value]="inputFieldValue" aria-autocomplete="list" [attr.aria-controls]="listId" role="searchbox" [attr.aria-expanded]="overlayVisible" aria-haspopup="true" [attr.aria-activedescendant]="'p-highlighted-option'"
+            [ngClass]="'ui-inputtext ui-widget ui-state-default ui-corner-all ui-autocomplete-input'" [value]="inputFieldValue" aria-autocomplete="list" [attr.aria-controls]="overlayVisible ? listId : null" role="list" 
[attr.aria-expanded]="overlayVisible" aria-haspopup="true" [attr.aria-activedescendant]="(overlayVisible && #panel.parentElement.querySelector('.ui-state-highlight')) ? 'p-highlighted-option' : null"
```
and
```
-                        (mouseenter)="highlightOption=option" (mouseleave)="highlightOption=null" [id]="highlightOption == option ? 'p-highlighted-option':''" (click)="selectItem(option)">
+                        (mouseenter)="highlightOption=option" (mouseleave)="highlightOption=null" [id]="highlightOption == option ? 'p-highlighted-option': null" (click)="selectItem(option)">
```

Hope this helps.

**Comment by cagataycivici** [comment]
Unable to replicate with PrimeNG 11.0.0-rc.1, if the issue persists please create a new ticket with a test case reproducing the issue e.g. stackblitz or a github repo and it will be reviewed by our team once again.

**Comment by StefanCardnell** [comment]
@cagataycivici @sandipchitale 

This ticket is valid. You can install the Chrome axe Devtools and run it on the show case for `p-autocomplete`:

https://www.primefaces.org/primeng/showcase/#/autocomplete

There are a couple critical issues which crop up for each autocomplete:

- Invalid ARIA attribute value: aria-activedescendant="p-highlighted-option"
- ARIA attribute is not allowed: aria-expanded="false"

The problem stems from the `role=searchbox` on the input. IMO the autocomplete more closely matches the definition of the `combobox` role and not `searchbox`.

https://www.w3.org/TR/wai-aria-1.1/#combobox
https://www.w3.org/TR/wai-aria-1.1/#searchbox

Changing the role to combobox fixes the `aria-expanded` issue.

The `aria-activedescendant` issue is fixable by only setting this attribute when the dialog is open and an option is highlighted.

Additionally, as per the combobox definition above, `aria-haspopup` should be set to `listbox` and not `true`.
